src/nitpick/style/fetchers/github.py

```diff
@@ -42,6 +51,21 @@ def default_branch(self) -> str:
         """
         return get_default_branch(self.api_url)
 
+    @property
+    def credentials(self) -> Tuple:
+        """Credentials encoded in this URL.
+
+        A tuple of ``(api_token, '')`` if present, or empty tuple otherwise.  If
+        the value of ``api_token`` begins with ``$``, it will be replaced with
+        the value of the environment corresponding to the remaining part of the
+        string.
```

Comment:
> Write documentation when there's a new API or functionality _suggestions where to document?_

I would write on `configuration.rst` at the bottom of the "Remote style" header.
Something like this, not necessarily with these exact wording:

```
If your style is on a private GitHub repo, you can provide the token directly on the URL.
Or you can use an environment variable to avoid keeping secrets in plain text.

.. code-block:: toml

    [tool.nitpick]
    # A literal token
    style = "github://p5iCG5AJuDgY@some-user/a-private-repo@some-branch/nitpick-style.toml"

    # Or reading the secret value from the MY_AUTH_KEY env var
    style = "github://$MY_AUTH_KEY@some-user/a-private-repo@some-branch/nitpick-style.toml"

.. note::

    Your token can't contain the `$` and `@` characters.

```

You could also use on the .rst file part of these sentences from the docstring above.

But you don't need to write the docs if you don't want to.
I can add a commit with documentation once this pull request is ready.